### ISSUE TITLE
Issues.list() does not work. XHR reference missing

### ISSUE THREAD

**Issue by harrisonhjones** [michael/github#288]
Hi everyone. I'm having a frustrating issue. When I try simply code like this:

```
var Github = require('github-api');
var ghUsername = 'michael';
var ghRepo = 'github';

var github = new Github({
  token: TOKEN,
  auth: "oauth"
});

var issues = github.getIssues(ghUsername, ghRepo);
issues.list(null, function(err, issueList) {
    console.log(err);
    console.log(issueList.length);
});
```
I get an empty console response. I've narrowed it down to [this line](https://github.com/michael/github/blob/master/src/github.js#L116) in github.js. It looks like `axios` no longer provides the xhr `request` in their response. The link header is obviously still there (can be verified by opening up [this link](https://api.github.com/repos/michael/github/issues?per_page=2) and viewing the headers) but we apparently cannot rely on having access to xhr.

I can propose a fix for this (with a PR) but I wanted someone to validate my findings first. 

Thanks,
Harrison

**Comment by AurelioDeRosa** [comment]
Hi.

axios never exposed the `XHR` object, that's why we're using our own fork. We do provide the `XHR` object as you can see on [this line](https://github.com/michael/github/blob/master/src/github.js#L89). The object is passed as the third argument since we implemented this feature.

**Comment by harrisonhjones** [comment]
When I go in and manually add a `console.log(response.request)` before calling `cb` I get `undefined`. I installed the module using `npm install --save github-api`. Did I do something wrong?

**Comment by harrisonhjones** [comment]
Note, I just tested my example on a clean install of github-api on a completely different computer and I have the same issue, no issues are returned.

**Comment by dominikschubert** [comment]
I'm experiencing the same problem with both basic auth and oauth. No issues are returned in the callback.
The package(github-api@0.11.2) was freshly installed via npm

**Comment by AurelioDeRosa** [comment]
I've spent a bit of time on this issue and I'm not able to replicate it. Our test for the `list()` method is passing and working correctly. The only bit I was able to find out is that the `_requestAllPages` method was not forwarding the `XHR` object to the callback method provided by the user. However, this has been fixed in commit 5bcfa70cc4817b9c1b7e771537cc7f6553dae9d9.

My suggestion would be to adapt our test for Issues to your case and see what happens, or to create a JSFiddle that reproduces the issue. I'm closing this issue for now.
